### ISSUE TITLE
inconsistent_digit_grouping doesn't like grouping 0s by base 2

### ISSUE THREAD

**Issue by gilescope** [rust-lang/rust-clippy#7230]
Lint name: inconsistent-digit-grouping

I tried this code:

```rust
1_0000_0000
```
The lint suggests the following grouping but it doesn't show the same intent:
```rust
100_000_000
```

Many times we're dealing with computers that like base 2. We should allow consistent base 2 groupings as not being something we should complain about.

This is inconsistent and I would be ok with it linting:
```
1_000_0_0000
```

But if there's a consistent pattern I think it's really not a helpful lint. I think we need to get a bit more nuanced with how this lint is designed.

**Comment by xFrednet** [comment]
Hey, thank you for the report! Could you try to reproduce this false positive again? I've tried the example in Rust's playground, but Clippy didn't trigger on `1_0000_000`. See [playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=fcde0a40b105212661c5af8332fc931b) :upside_down_face:

**Comment by samueltardieu** [comment]
I propose we close this one (no activity in 3.5 years, no confirmed bug, no reproducer), let's reopen it with an example if needed.
